pkg/utils/slices.go

```diff
@@ -0,0 +1,16 @@
+package utils
+
+import (
+	"strings"
+)
+
+func Intersect(this, that []string) bool {
```

Comment:
I saw that it's used only for sources. What's more sources are used only with this method. Can we just create a type with our method?

```go

type Sources map[string]struct{}

func (s Sources) Intersect(with []string) bool {
	if s == nil {
		return false
	}
	
	for _, name := range with {
		if _, found := s[strings.ToLower(name)]; found {
			return true
		}
	}
	return false
}
```

a small benefit is that we would not require 2 `for` loops

Reply:
I see what you mean. 🙂

But given the massive changes in this PR, I'm happy to take this in a future refactoring - if that's ok.